fontir/src/ir.rs

```diff
@@ -809,7 +796,60 @@ impl NameBuilder {
         self.names
     }
 
+    pub fn get_fallback_or_default(&self, name_id: NameId, fallbacks: &[NameId]) -> Option<&str> {
+        fallbacks
+            .iter()
+            .find_map(|n| {
+                let key = self.name_to_key.get(n)?;
+                self.names.get(key).map(|s| s.as_str())
+            })
+            .or_else(|| default_value(name_id))
+    }
+
     pub fn apply_default_fallbacks(&mut self, vendor_id: &str) {
+        // If the legacy subfamily isn't set, we fall back to the typographic subfamily;
+        // but because the spec recommends the former to only contain "Regular",
+        // "Bold", "Italic", or "Bold Italic", if this fallback isn't RIBBI already,
+        // we append it to the legacy family name and set the legacy subfamily to the
+        // default "Regular", the same way fontmake does via ufo2ft.
+
+        // https://github.com/googlefonts/ufo2ft/blob/bb79cae53f1c160c7174ebef0d463c7a28a7552a/Lib/ufo2ft/fontInfoData.py#L76
+        let family_suffix = if self.contains_key(NameId::SUBFAMILY_NAME) {
+            // assume this is good, no need to add suffix to family name.
+            None
+        } else {
+            let fallback_subfamily = self
+                .get_fallback_or_default(
+                    NameId::SUBFAMILY_NAME,
+                    &[NameId::TYPOGRAPHIC_SUBFAMILY_NAME],
+                )
+                .unwrap() // safe because SUBFAMILY_NAME has a default_value
+                .to_string();
+
+            let (fallback_subfamily, family_suffix) = if is_ribbi(&fallback_subfamily) {
+                (fallback_subfamily, None)
+            } else {
+                ("Regular".to_string(), Some(fallback_subfamily))
+            };
+
+            self.add(NameId::SUBFAMILY_NAME, fallback_subfamily);
+
+            family_suffix
+        };
```

Comment:
This feels ... more complicated than the work being done but I'm not immediately sure how to fix it.

Adding a helper for the get/unwrap/to-string helps a little maybe.

```rust
    /// Fetch a fallback that definitely has a value in a String to avoid an immutable borrow on self
    /// Only safe to use with names that have default values
    fn fallback_string(&mut self, name_id: NameId, fallback: NameId) -> String {
        self.get_fallback_or_default(name_id, &[fallback]).unwrap().to_string()
    }
```

Optional given I have no real suggestion for improvement.